guvnor-ala/guvnor-ala-docker-provider/src/main/java/org/guvnor/ala/docker/executor/DockerProviderConfigExecutor.java

```diff
@@ -29,23 +31,25 @@
 import org.guvnor.ala.runtime.providers.ProviderDestroyer;
 import org.guvnor.ala.runtime.providers.ProviderId;
 
-public class DockerProviderConfigExecutor implements ProviderBuilder<DockerProviderConfig, DockerProvider>,
-                                                     ProviderDestroyer,
-                                                     FunctionConfigExecutor<DockerProviderConfig, DockerProvider> {
+public class DockerProviderConfigExecutor
+        implements ProviderBuilder<DockerProviderConfig, DockerProvider>,
+                   ProviderDestroyer,
+                   FunctionConfigExecutor<DockerProviderConfig, DockerProvider> {
 
     private RuntimeRegistry runtimeRegistry;
 
     @Inject
-    public DockerProviderConfigExecutor( final RuntimeRegistry runtimeRegistry ) {
+    public DockerProviderConfigExecutor(final RuntimeRegistry runtimeRegistry) {
         this.runtimeRegistry = runtimeRegistry;
     }
 
     @Override
-    public Optional<DockerProvider> apply( final DockerProviderConfig dockerProviderConfig ) {
-        final DockerProviderImpl provider = new DockerProviderImpl( dockerProviderConfig.getName(), 
-                dockerProviderConfig.getHostIp(), dockerProviderConfig );
-        runtimeRegistry.registerProvider( provider );
-        return Optional.of( provider );
+    public Optional<DockerProvider> apply(final DockerProviderConfig dockerProviderConfig) {
+        final DockerProviderImpl provider = new DockerProviderImpl(
+                new DockerProviderConfigImpl(dockerProviderConfig.getName(),
```

Comment:
I'm not the creator of this DockerXXXStuff in geneal, and the only change I did in this class applying formating.
But going back to your query, the parameter DockerProviderConfig dockerProviderConfig is just an interface, and DockerProviderConfig is not a ClonableConfig, so unless starting to do castings to see if it's ClonableConfig, etc, the cloning can't be directly applied.
Since this code was left in this way by the creator, etc, and it's not the priority right now, I'd prefer to not touch it.